src/v0/destinations/tiktok_ads/transformV2.js

```diff
@@ -40,7 +40,7 @@ const getTrackResponsePayload = (message, destConfig, event) => {
   }
 
   // if contents is not present but we have properties.products present which has fields with superset of contents fields
-  if (payload.properties && !payload.properties.contents && message.properties.products) {
+  if (!payload.properties?.contents && message.properties?.products) {
```

Comment:
[Q] Should we do `Array.isArray(message?.properties?.products)` instead ? This would make sure products is an array if it exists.

Reply:
We are already check if product is array or not in getContents util
https://github.com/rudderlabs/rudder-transformer/blob/f7ec0a1244a7b97e6b40de5ed9881c63300866dc/src/v0/destinations/tiktok_ads/util.js#L14